rasa_sdk/forms.py

```diff
@@ -689,3 +689,76 @@ def _get_entity_type_of_slot_to_fill(
                 return None
 
         return entity_type
+
+
+class FormValidationAction(Action):
+    """An action that validates if every extracted slot is valid."""
+
+    def name(self) -> Text:
+        """Unique identifier of this action.
+
+        Returns:
+            Name of the action.
+        """
+
+        raise NotImplementedError("An action must implement a name")
+
+    async def validate(
+        self,
+        dispatcher: "CollectingDispatcher",
+        tracker: "Tracker",
+        domain: "DomainDict",
+    ) -> List[EventType]:
+        """Validate slots by calling a validation function for each slot.
+
+        Args:
+            dispatcher: the dispatcher which is used to
+                send messages back to the user.
+            tracker: the state tracker for the current user.
+            domain: the bot's domain.
+        Returns:
+            A dictionary of `rasa_sdk.events.Event` instances.
+        """
+        slots: Dict[Text, Any] = tracker.form_slots_to_validate()
+
+        for slot_name, slot_value in slots.items():
+            function_name = f"validate_{slot_name}"
+            validate_func = getattr(self, function_name, lambda *x: False)
+
+            if utils.is_coroutine_action(validate_func):
+                validation_output = await validate_func(
+                    slot_value, dispatcher, tracker, domain
+                )
+            else:
+                validation_output = validate_func(
+                    slot_value, dispatcher, tracker, domain
+                )
+
+            if validation_output:
+                slots.update(validation_output)
+            else:
+                warnings.warn(
+                    f"Cannot validate `{slot_name}`: make sure the validation function is specified and "
+                    f"returns a list of `rasa_sdk.events.Event` instances."
+                )
+
+        return [SlotSet(slot, value) for slot, value in slots.items()]
+
+    async def run(
+        self,
+        dispatcher: "CollectingDispatcher",
+        tracker: "Tracker",
+        domain: "DomainDict",
+    ) -> List[EventType]:
+        """Execute the side effects of this action.
+
+        Args:
+            dispatcher: the dispatcher which is used to
+                send messages back to the user.
+            tracker: the state tracker for the current user.
+            domain: the bot's domain.
+        Returns:
+            A dictionary of `rasa_sdk.events.Event` instances.
```

Comment:
This is a list of events. Imo you don't have to specify a docstring as the docstring is already specified in the parent class.